src/Voron/Impl/FreeSpace/FreeSpaceHandling.cs

```diff
@@ -489,7 +491,13 @@ public void FreePage(LowLevelTransaction tx, long pageNumber)
                 {
                     sba = !result.HasValue ? new StreamBitArray() : new StreamBitArray(result.CreateReader().Base);
                 }
-                sba.Set((int)(pageNumber % NumberOfPagesInSection), true);
+
+                var index = (int)(pageNumber % NumberOfPagesInSection);
+
+                if (sba.Get(index))
+                    throw new InvalidOperationException($"trying to double free page {pageNumber}");
```

Comment:
Should this be in debug mode?
Or should we add a `Set()` the verify that the previous value was `false` ?

Reply:
yeah, we can do something like this
```
    public void Set(int index, bool value)
    {
        var current = Get(index);
        if (current == value)
            ThrowValueAlreadyExists(index, value);;
      ...
```